automation/validate-pvc-name-stability.py

```diff
@@ -0,0 +1,92 @@
+#! python
+
+import logging
+import os
+import os.path
+import yaml
+import sys
+
+from kubernetes import client, config
+from openshift.dynamic import DynamicClient
+
+
+def validatePVCNames(path, liveTemplates):
+    templates = [f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f))]
+    errors = []
+    for templateFilename in templates:
+        with open(os.path.join(path, templateFilename), 'r') as stream:
+            try:
+                template = yaml.safe_load(stream)
+            except yaml.YAMLError as exc:
+                raise exc
+
+            if template is None:
+                logging.info("Empty template file: %s", template)
+                continue
+
+            templateName = template["metadata"]["name"]
+            logging.info("Checking PVC name stability for: {}".format(templateName))
+
+            pvcName = getPVCNameFrom(template)
+            pvcNamespace = getPVCNamespaceFrom(template)
+
+            matchingLiveTemplate = liveTemplates.get(templateName)
+            if matchingLiveTemplate:
+                liveTemplatePVCName = getPVCNameFrom(matchingLiveTemplate)
+                liveTemplatePVCNamespace = getPVCNamespaceFrom(matchingLiveTemplate)
+
+                if pvcName != liveTemplatePVCName:
+                    errors.append("PVC name: {} was modified in: {}".format(pvcName, templateName))
+                if pvcNamespace != liveTemplatePVCNamespace:
+                    errors.append("PVC namespace: {} was modified in: {}".format(pvcNamespace, templateName))
+            else:
+                logging.info("Missing liveTemplate for {}".format(templateName))
+
+    if len(errors) > 0:
```

Comment:
This could be reduced to `if errors:` unless we want to also check the instance type.

Reply:
That's what you get from mostly only doing go for 4 years.